### PR TITLE
fix(curriculum): clean-up Project Euler 101-120

### PR DESCRIPTION
**Pull request by gikf**
- Cleans description.
- Changes generic `eulerXXX` function name.
- Adds missing test data.
- Adds placeholders for images.
- Uses MathJax, a lot.
- Related to #40896.
- Challenge 102 is already reworked and didn't need clean-up.

Should images be committed to https://github.com/freeCodeCamp/cdn? Other Project Euler challenges having images seem to have them somewhere else, ie. https://cdn-media-1.freecodecamp.org/project-euler/right-triangles-integer-coordinates-2.png

- [x] Add images
- [x] Merge https://github.com/freeCodeCamp/cdn/pull/75

Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

### CODE REVIEW

**Comment by nhcarrigan** [comment]
> Should images be committed to https://github.com/freeCodeCamp/cdn? Other Project Euler challenges having images seem to have them somewhere else, ie. https://cdn-media-1.freecodecamp.org/project-euler/right-triangles-integer-coordinates-2.png

Those are on our S3 instance, I believe. @raisedadead would you rather these images go on S3 or on our CDN repo?

**Comment by raisedadead** [comment]
Sorry for the delay in the reply, I think we should put all the curriculum-related images on the CDN repo.

File: curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-109-darts.md [hunk 1]
```diff
@@ -10,41 +10,51 @@ dashedName: problem-109-darts
...
+p109.png
 
-D3
+The score of a dart is determined by the number of the region that the dart lands in. A dart landing outside the red/green outer ring scores zero. The black and cream regions inside this ring represent single scores. However, the red/green outer ring and middle ring score double and treble scores respectively.
 
-D1 D2
+At the centre of the board are two concentric circles called the bull region, or bulls-eye. The outer bull is worth 25 points and the inner bull is a double, worth 50 points.
```
Comment (Sembauke) [inline curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-109-darts.md:17]:
```suggestion
At the center of the board are two concentric circles called the bull region, or bulls-eye. The outer bull is worth 25 points and the inner bull is a double, worth 50 points.
```

File: curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-109-darts.md [hunk 2]
```diff
@@ -10,41 +10,51 @@ dashedName: problem-109-darts
...
 
-S2 D2
+There are many variations of rules but in the most popular game the players will begin with a score 301 or 501 and the first player to reduce their running total to zero is a winner. However, it is normal to play a "doubles out" system, which means that the player must land a double (including the double bulls-eye at the centre of the board) on their final dart to win; any other dart that would reduce their running total to one or lower means the score for that set of three darts is "bust".
```
Comment (Sembauke) [inline curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-109-darts.md:19]:
```suggestion
There are many variations of rules but in the most popular game the players will begin with a score of 301 or 501 and the first player to reduce their running total to zero is a winner. However, it is normal to play a "doubles out" system, which means that the player must land a double (including the double bulls-eye at the center of the board) on their final dart to win; any other dart that would reduce their running total to one or lower means the score for that set of three darts is "bust".
```

File: curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-109-darts.md [hunk 3]
```diff
@@ -10,41 +10,51 @@ dashedName: problem-109-darts
...
+  S4        & D1 &    \\\\
+  S1        & S1 & D2 \\\\
+  S1        & T1 & D1 \\\\
+  S1        & S3 & D1 \\\\
+  D1        & D1 & D1 \\\\
+  D1        & S2 & D1 \\\\
+  S2        & S2 & D1
+\end{array}$$
 
 Note that D1 D2 is considered different to D2 D1 as they finish on different doubles. However, the combination S1 T1 D1 is considered the same as T1 S1 D1. In addition we shall not include misses in considering combinations; for example, D3 is the same as 0 D3 and 0 0 D3. Incredibly there are 42336 distinct ways of checking out in total. How many distinct ways can a player checkout with a score less than 100?
```
Comment (Sembauke) [inline curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-109-darts.md:37]:
```suggestion
Note that D1 D2 is considered different from D2 D1 as they finish on different doubles. However, the combination S1 T1 D1 is considered the same as T1 S1 D1. In addition, we shall not include misses in considering combinations; for example, D3 is the same as 0 D3 and 0 0 D3. Incredibly there are 42336 distinct ways of checking out in total. How many distinct ways can a player checkout with a score less than 100?
```

File: curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-116-red-green-or-blue-tiles.md [hunk 4]
```diff
@@ -12,31 +12,39 @@ A row of five black square tiles is to have a number of its tiles replaced with
...
 If green tiles are chosen there are three ways.
 
+p116_2.png
+
 And if blue tiles are chosen there are two ways.
 
-Assuming that colours cannot be mixed there are 7 + 3 + 2 = 12 ways of replacing the black tiles in a row measuring five units in length. How many different ways can the black tiles in a row measuring fifty units in length be replaced if colours cannot be mixed and at least one coloured tile must be used? NOTE: This is related to Problem 117.
+p116_3.png
+
+Assuming that colours cannot be mixed there are 7 + 3 + 2 = 12 ways of replacing the black tiles in a row measuring five units in length. How many different ways can the black tiles in a row measuring fifty units in length be replaced if colours cannot be mixed and at least one coloured tile must be used?
```
Comment (Sembauke) [inline curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-116-red-green-or-blue-tiles.md:25]:
```suggestion
Assuming that colors cannot be mixed there are 7 + 3 + 2 = 12 ways of replacing the black tiles in a row measuring five units in length. How many different ways can the black tiles in a row measuring fifty units in length be replaced if colors cannot be mixed and at least one colored tile must be used?
```

**Review by Sembauke — CHANGES_REQUESTED** [review]

**Comment by gikf** [comment]
Blocked waiting for https://github.com/freeCodeCamp/cdn/pull/75

**Review by raisedadead — APPROVED** [review]
The CDN PR has been merged.

**Review by Sembauke — APPROVED** [review]

**Pull request merged**